src/Nethermind/Nethermind.Merge.Plugin/Handlers/ForkchoiceUpdatedHandler.cs

```diff
@@ -250,6 +266,29 @@ public Task<ResultWrapper<ForkchoiceUpdatedV1Result>> Handle(ForkchoiceStateV1 f
         return null;
     }
 
+    private async Task<BlockHeader?> FetchHeaderFromPeer(ForkchoiceStateV1 forkchoiceState)
+    {
+        using CancellationTokenSource cts = new CancellationTokenSource();
+        cts.CancelAfter(TimeSpan.FromSeconds(2));
+
+        BlockHeader[]? headers = null;
+        try
+        {
+            headers = await _syncPeerPool.AllocateAndRun(
+                peer => peer.GetBlockHeaders(forkchoiceState.HeadBlockHash, 1, 0, cts.Token),
+                BySpeedStrategy.FastestHeader,
+                AllocationContexts.Headers,
+                cts.Token);
+        }
+        catch (OperationCanceledException)
+        {
+            // Timeout or no peer.
+            return null;
+        }
+
+        return headers?.Length == 1 ? headers[0] : null;
+    }
+
```

Comment:
Make this extension method on `_syncPeerPool`, pass only `forkchoiceState.HeadBlockHash` instead of full `ForkchoiceStateV1`

Reply:
Done